pkg/resource/table/hooks_replica_updates.go

```diff
@@ -0,0 +1,459 @@
+	if err != nil {
+		// Handle specific errors
+		if awsErr, ok := ackerr.AWSError(err); ok {
+			// Handle ValidationException - when replicas are not part of the global table
+			if awsErr.ErrorCode() == "ValidationException" &&
+				strings.Contains(awsErr.ErrorMessage(), "not part of the global table") {
+				// A replica was already deleted
+				rlog.Debug("replica already deleted from global table, will requeue",
+					"table", *latest.ko.Spec.TableName,
+					"error", awsErr.ErrorMessage())
+				return ackrequeue.NeededAfter(
+					ErrTableUpdating,
+					60*time.Second,
+				)
+			}
+
+			// Handle ResourceInUseException - when the table is being updated
+			if awsErr.ErrorCode() == "ResourceInUseException" {
+				rlog.Debug("table is currently in use, will retry",
+					"table", *latest.ko.Spec.TableName,
+					"error", awsErr.ErrorMessage())
+				return ackrequeue.NeededAfter(
+					ErrTableUpdating,
+					30*time.Second,
+				)
+			}
+			return err
+		}
+	}
+
+	// If there are more replicas to process, requeue
+	if replicasInQueue > 0 {
+		rlog.Debug("more replica updates pending, will requeue",
+			"table", *latest.ko.Spec.TableName,
+			"remaining_updates", replicasInQueue)
+		return requeueWaitWhileUpdating
+	}
+
+	return nil
+}
+
+// newUpdateTableReplicaUpdatesOneAtATimePayload creates the UpdateTable input payload for replica updates,
+// processing only one replica at a time
+func (rm *resourceManager) newUpdateTableReplicaUpdatesOneAtATimePayload(
+	ctx context.Context,
+	latest *resource,
+	desired *resource,
+) (input *svcsdk.UpdateTableInput, replicasInQueue int, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.newUpdateTableReplicaUpdatesOneAtATimePayload")
+	defer func() {
+		exit(err)
+	}()
+
+	createReplicas, updateReplicas, deleteRegions := calculateReplicaUpdates(latest, desired)
+
+	input = &svcsdk.UpdateTableInput{
+		TableName:      aws.String(*desired.ko.Spec.TableName),
+		ReplicaUpdates: []svcsdktypes.ReplicationGroupUpdate{},
+	}
+
+	totalReplicasOperations := len(createReplicas) + len(updateReplicas) + len(deleteRegions)
+	replicasInQueue = totalReplicasOperations - 1
+
+	// Process replica updates in order: create, update, delete
+	// We'll only perform one replica action at a time
+
+	if len(createReplicas) > 0 {
+		replica := *createReplicas[0]
+		rlog.Debug("creating replica in region", "table", *desired.ko.Spec.TableName, "region", *replica.RegionName)
+		input.ReplicaUpdates = append(input.ReplicaUpdates, createReplicaUpdate(createReplicas[0]))
+		return input, replicasInQueue, nil
+	}
+
+	if len(updateReplicas) > 0 {
+		replica := *updateReplicas[0]
+		rlog.Debug("updating replica in region", "table", *desired.ko.Spec.TableName, "region", *replica.RegionName)
+		updateReplica := updateReplicaUpdate(updateReplicas[0])
+		if updateReplica.Update != nil {
+			input.ReplicaUpdates = append(input.ReplicaUpdates, updateReplica)
+		}
```

Comment:
```suggestion
		replica := updateReplicas[0]
		rlog.Debug("updating replica in region", "table", *desired.ko.Spec.TableName, "region", *replica.RegionName)
		updateReplica := updateReplicaUpdate(replica)
		if updateReplica.Update != nil {
			input.ReplicaUpdates = append(input.ReplicaUpdates, updateReplica)
		}
```
also: `updateReplicaUpdate` isn't super clear on what it's really doing.

Reply:
creates the update payload with the caveat that updates to just replica GSI without the any other parameter changes are ignored since they are invalid
Replica's GSI is updated by the table update.